File: VB/09_Interaction/AddTextInHeaderAndFooter.vb
Imports System.ComponentModel
Imports System.Text
Imports Spire.Pdf
Imports Spire.Pdf.Graphics

Namespace AddTextInHeaderAndFooter
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub button1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles button1.Click
			Dim path As String = "..\..\..\..\..\..\Data\"
			'pdf file 
			Dim input As String = path & "Sample4.pdf"

			'open pdf document
			Dim doc As New PdfDocument(input)

			Dim brush As PdfBrush = PdfBrushes.Black
			Dim pen As New PdfPen(brush, 0.75f)
			Dim font As New PdfTrueTypeFont(New Font("Arial", 9f, FontStyle.Italic), True)
			Dim leftAlign As New PdfStringFormat(PdfTextAlignment.Left)
			Dim rightAlign As New PdfStringFormat(PdfTextAlignment.Right)
			leftAlign.MeasureTrailingSpaces = True
			rightAlign.MeasureTrailingSpaces = True
			Dim margin As PdfMargins = doc.PageSettings.Margins

			Dim space As Single = font.Height * 0.75f
			Dim x As Single = 0
			Dim y As Single = 0
			Dim width As Single = 0

			For Each page As PdfPageBase In doc.Pages
				page.Canvas.SetTransparency(0.5f)
				x = margin.Left
				width = page.Canvas.ClientSize.Width - margin.Left - margin.Right
				y = margin.Top - space
				page.Canvas.DrawLine(pen, x, y, x + width, y)
				y = y - 1 - font.Height

				'draw header text into page
				page.Canvas.DrawString("Demo of Spire.Pdf", font, brush, x, y, leftAlign)

				brush = PdfBrushes.Blue
				font = New PdfTrueTypeFont(New Font("Arial", 8f, FontStyle.Regular), True)
				y = page.Canvas.ClientSize.Height - margin.Bottom - font.Height

				'draw footer text into page
				page.Canvas.DrawString("Created by E-iceblue Co,.Ltd", font, brush, x + width, y, rightAlign)

				page.Canvas.SetTransparency(1)
			Next page

			Dim output As String = "AddTextInHeaderAndFooter.pdf"

			'Save pdf file.
			doc.SaveToFile(output)
			doc.Close()

			'Launching the Pdf file.
			PDFDocumentViewer(output)
		End Sub
		Private Sub PDFDocumentViewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub
	End Class
End Namespace

File: CS/09_Interaction/AddTextInHeaderAndFooter.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Pdf;
using Spire.Pdf.Graphics;

namespace AddTextInHeaderAndFooter
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            string path = "..\\..\\..\\..\\..\\..\\Data\\";
            //pdf file 
            string input = path + "Sample4.pdf";

            //open pdf document
            PdfDocument doc = new PdfDocument(input);

            PdfBrush brush = PdfBrushes.Black;
            PdfPen pen = new PdfPen(brush, 0.75f);
            PdfTrueTypeFont font = new PdfTrueTypeFont(new Font("Arial", 9f, FontStyle.Italic), true);
            PdfStringFormat leftAlign = new PdfStringFormat(PdfTextAlignment.Left);
            PdfStringFormat rightAlign = new PdfStringFormat(PdfTextAlignment.Right);
            leftAlign.MeasureTrailingSpaces = true;
            rightAlign.MeasureTrailingSpaces = true;
            PdfMargins margin = doc.PageSettings.Margins;

            float space = font.Height * 0.75f;
            float x = 0;
            float y = 0;
            float width = 0;

            foreach (PdfPageBase page in doc.Pages)
            {
                page.Canvas.SetTransparency(0.5f);
                x = margin.Left;
                width = page.Canvas.ClientSize.Width - margin.Left - margin.Right;
                y = margin.Top - space;
                page.Canvas.DrawLine(pen, x, y, x + width, y);
                y = y - 1 - font.Height;

                //draw header text into page
                page.Canvas.DrawString("Demo of Spire.Pdf", font, brush, x, y, leftAlign);

                brush = PdfBrushes.Blue;
                font = new PdfTrueTypeFont(new Font("Arial", 8f, FontStyle.Regular), true);
                y = page.Canvas.ClientSize.Height - margin.Bottom - font.Height;
               
                //draw footer text into page
                page.Canvas.DrawString("Created by E-iceblue Co,.Ltd", font, brush, x + width, y, rightAlign);

                page.Canvas.SetTransparency(1);
            }

            string output = "AddTextInHeaderAndFooter.pdf";

            //Save pdf file.
            doc.SaveToFile(output);
            doc.Close();

            //Launching the Pdf file.
            PDFDocumentViewer(output);
        }
        private void PDFDocumentViewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }
    }
}
